src/workloads/streams/LargeHoppingWindow.yml

```diff
@@ -0,0 +1,183 @@
+    channel: *Channel
+    url: *Url
+    dateTime: "2023-01-01T00:00:05.000"
+
+  Batch1000x: &Batch1000x {^Array: { of: *Document, number: 1000 }}
+
+Actors:
+- Name: Setup
+  Type: RunCommand
+  ClientName: Stream
+  Threads: 1
+  Phases:
+  - Phase: 0
+    Repeat: 1
+    Database: *DatabaseName
+    Operations:
+    - OperationMetricsName: CreateStreamProcessor
+      OperationName: RunCommand
+      OperationCommand:
+        streams_startStreamProcessor: ""
+        name: *StreamProcessorName
+        pipeline: [
+          {
+            $source: {
+              connectionName: "__testMemory",
+              timeField: { $convert: { input: "$dateTime", to: "date" }},
+              allowedLateness: { size: 1, unit: "second" },
+            }
+          },
+          {
+            $hoppingWindow: {
+              interval: { size: 1, unit: "second" },
+              hopSize: { size: 250, unit: "ms" },
+              pipeline: [
+                { 
+                  $group: {
+                    _id: {
+                      auction: "$auction",
+                      url: "$url"
+                    },
+                    minPrice: { $min: "$price" },
+                    maxPrice: { $max: "$price" },
+                    sumPrice: { $sum: "$price" },
+                    avgPrice: { $avg: "$price" }
+                  }
+                }
+              ]
+            }
+          },
+          { $emit: { connectionName: "__noopSink" } }
+        ]
+        connections: [{ name: "__testMemory", type: "in_memory", options: {} }]
+  - Phase: 1..2
+    Nop: true
+  - Phase: 3
+    Repeat: 1
+    Database: *DatabaseName
+    Operations:
+    - OperationMetricsName: Stop
+      OperationName: RunCommand
+      OperationCommand:
+        streams_stopStreamProcessor: ""
+        name: *StreamProcessorName
+
+- Name: Insert_Batch1000x
+  Type: RunCommand
+  ClientName: Stream
+  Threads: *NumThreads
+  Phases:
+  - Phase: 0
+    Nop: true
+  - Phase: 1
+    Repeat: *NumBatch1000xPerThread
+    Database: *DatabaseName
+    Operations:
+    - OperationMetricsName: Insert
+      OperationName: RunCommand
+      OperationCommand:
+        streams_testOnlyInsert: ""
+        name: *StreamProcessorName
+        documents: *Batch1000x
+  - Phase: 2
+    Repeat: 1
+    Database: *DatabaseName
+    Operations:
+    - OperationMetricsName: Insert_Flush
+      OperationName: RunCommand
+      OperationCommand:
+        streams_testOnlyInsert: ""
+        name: *StreamProcessorName
+        documents:
+        - *FlushDocument
+  - Phase: 3
+    Nop: true
+
+- Name: LargeHoppingWindow.MemorySource.Stats.Insert
+  Type: StreamStatsReporter
+  ClientName: Stream
+  Database: *DatabaseName
+  Threads: 1
+  Phases:
+  - Phase: 0
+    Nop: true
+  - Phase: 1
+    Repeat: 1
+    StreamProcessorName: *StreamProcessorName
+    ExpectedDocumentCount: *ExpectedDocumentCount
+  - Phase: 2..3
+    Nop: true
+
+- Name: LargeHoppingWindow.MemorySource.Stats.Flush
```

Comment:
Is this the stage that waits for all the processing of in-memory source documents to finish?

Reply:
yup, this is what will periodically poll `streams_getStats` until `inputDocumentCount` from the stats response hits the  `ExpectedDocumentCount`. This is also the main actor that we want to track for throughput (specifically stats emitted from phase 2 of this actor) because those will be the synthetic measurements for throughput which it'll report periodically (every one second) based on `streams_getStats` response.